httpx/dispatch/connection.py

```diff
@@ -92,13 +88,7 @@ def __init__(
         http_version = stream.get_http_version()
         logger.trace(f"connected http_version={http_version!r}")
 
-        if http_version == "HTTP/2":
-            self.h2_connection = HTTP2Connection(
-                stream, backend=self.backend, on_release=on_release
-            )
-        else:
-            assert http_version == "HTTP/1.1"
-            self.h11_connection = HTTP11Connection(stream, on_release=on_release)
+        self.set_open_connection(http_version, socket=stream, on_release=on_release)
```

Comment:
Note that we've got an outstanding `stream` vs `socket` instance here (following up #601). Wondering if we'd want to push the renaming further with `.open_tcp_socket()` and `.open_uds_socket()` on the `ConcurrencyBackend` interface.

Reply:
Seems reasonable, yup.